src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt

```diff
@@ -47,14 +45,8 @@ class IncentiveReviewsService(
     val deferredPositiveCaseNotesInLast3Months = async { getCaseNoteUsage("POS", "IEP_ENC", prisonerNumbers) }
     val deferredNegativeCaseNotesInLast3Months = async { getCaseNoteUsage("NEG", "IEP_WARN", prisonerNumbers) }
 
-    val bookingIds = offenders.content.map(OffenderSearchPrisoner::bookingId)
-    val nextReviewDates = getNextReviewDatesForOffenders(bookingIds)
-
-    val bookingIdsMissingIepReviews = bookingIds.toSet() subtract nextReviewDates.keys
-    if (bookingIdsMissingIepReviews.isNotEmpty()) {
-      throw ListOfDataNotFoundException(bookingIdsMissingIepReviews)
-    }
-    val overdueCount = nextReviewDates.map { r -> r.value.nextReviewDate }.count { it.isBefore(LocalDate.now(clock)) }
+    val nextReviewDates = nextReviewDateGetterService.getMany(offenders.content)
```

Comment:
Well, if for whatever reason it fails it wouldn't get to to this point but assuming the getter does not fail (e.g. no DB query errors etc) it would return a `Map` with all the dates for all the passed offenders.

That's why:
1. I've removed this exception
2. I've retained `!!` when I'm getting data out of the map, as `nextReviewDates[bookingId]` should always return a `LocalDate` for all the offenders